### PR TITLE
add python solution for mountain view (896)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-896.mdx [hunk 1]
```diff
@@ -125,4 +125,33 @@ class mountain implements Comparable<mountain> {
 
 </JavaSection>
 
+<PySection>
+```python
+import sys
+
+sys.stdin = open("mountains.in", "r")
+sys.stdout = open("mountains.out", "w")
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:133]: (optional) could you use `with` instead?

Comment (jjliewie) [reply solutions/usaco-896.mdx:133]: okay ill change all of them

**Comment by SansPapyrus683** [comment]
also add type annotations for the functions you use

File: solutions/usaco-896.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx]: changed it

File: solutions/usaco-896.mdx [hunk 2]
```diff
@@ -125,4 +125,31 @@ class mountain implements Comparable<mountain> {
...
+res = 1
+
+for i in range(1, n):
+	end = use[i][1]
+	if end > cur:
+		cur = end
+		res += 1
+
+with open('mountains.out', 'w') as out:
+	print(res, file=out)
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:151]:
```suggestion
print(res, file=open('mountains.out', 'w'))
```
up to you tho i feel like this looks so much nicer

File: solutions/usaco-896.mdx [hunk 3]
```diff
@@ -125,4 +125,31 @@ class mountain implements Comparable<mountain> {
 
 </JavaSection>
 
+<PySection>
+	
+```python
+use = []
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:131]: use? use what?

File: solutions/usaco-896.mdx [hunk 4]
```diff
@@ -125,4 +125,31 @@ class mountain implements Comparable<mountain> {
...
+with open('mountains.in') as read:
+	n = int(read.readline())
+	for _ in range(n):
+		a, b = map(int, read.readline.split())
+		use.append((a - b, a + b))
+
+use.sort(key=lambda x: (x[0], -x[1]))
+cur = use[0][1]
+
+res = 1
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:142]: either use `ans` like the other sols or just use a more informative var name

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-896.mdx (hunk 4 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx:142]: res stands for result? I'll change it to ans

File: solutions/usaco-896.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx:131]: yep ill change that too

File: solutions/usaco-896.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-896.mdx:142]: yeah but the other sols use `ans`

Comment (jjliewie) [reply solutions/usaco-896.mdx:142]: i will change it for you. <3

File: solutions/usaco-896.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx]: done

File: solutions/usaco-896.mdx (hunk 4 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx]: done

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: solutions/usaco-896.mdx [hunk 5]
```diff
@@ -125,4 +125,28 @@ class mountain implements Comparable<mountain> {
...
+	n = int(read.readline())
+	for _ in range(n):
+		a, b = map(int, read.readline().split())
+		pos.append((a - b, a + b))
+
+pos.sort(key=lambda x: (x[0], -x[1]))
+cur = pos[0][1]
+
+ans = 1
+for i in range(1, n):
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:142]: comment
"loop through all mountains, doing x..."

File: solutions/usaco-896.mdx [hunk 6]
```diff
@@ -125,4 +125,28 @@ class mountain implements Comparable<mountain> {
...
+	
+```python
+pos = []
+with open('mountains.in') as read:
+	n = int(read.readline())
+	for _ in range(n):
+		a, b = map(int, read.readline().split())
+		pos.append((a - b, a + b))
+
+pos.sort(key=lambda x: (x[0], -x[1]))
```
[the remark this answers is not in the document]
Comment (jjliewie) [reply solutions/usaco-896.mdx:138]: I'm sorting positions. idk man comments are so hard ill try

File: solutions/usaco-896.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-896.mdx:142]: oo right

File: solutions/usaco-896.mdx [hunk 7]
```diff
@@ -135,6 +135,7 @@ with open('mountains.in') as read:
 		a, b = map(int, read.readline().split())
 		pos.append((a - b, a + b))
 
+# sort mountain positions
```
Comment (SansPapyrus683) [inline solutions/usaco-896.mdx:138]: ...well of course you're sorting positions
what are you sorting them _by_?

Comment (jjliewie) [reply solutions/usaco-896.mdx:138]: okay. tbh i forgot what this problem was about i'll resolve and get back to u

**Comment by SansPapyrus683** [comment]
nononononon

**Comment by jjliewie** [comment]
> nononononon

tests... i'll work on the prs sometime this week after everything is over

**Pull request merged**